Runtime/Scripts/MobiledgeXIntegrationHelper.cs

```diff
@@ -227,16 +228,21 @@ public async Task<bool> FindCloudlet(string dmeHost = null, uint dmePort = 0)
                 {
                     throw new FindCloudletException("FindCloudletReply returned null. Make Sure you created App Instances for your Application and they are deployed in the correct region.");
                 }
-                if (reply.status != FindCloudletReply.FindStatus.FIND_FOUND)
+                if (reply.Status != FindCloudletReply.Types.FindStatus.FindFound)
                 {
-                    throw new FindCloudletException("Unable to findCloudlet. Status is " + reply.status);
+                    throw new FindCloudletException("Unable to findCloudlet. Status is " + reply.Status);
                 }
             }
 
-            Debug.Log("FindCloudlet with DME result: " + reply.status);
+            Debug.Log("FindCloudlet with DME result: " + reply.Status);
             latestFindCloudletReply = reply;
-            latestAppPortList = reply.ports;
-            return reply.status == FindCloudletReply.FindStatus.FIND_FOUND;
+            int porti = 0;
+            latestAppPortList = new AppPort[reply.Ports.Count];
+            foreach (var aport in reply.Ports)
+            {
+               latestAppPortList[porti++] = aport;
+            }
+            return reply.Status == FindCloudletReply.Types.FindStatus.FindFound;
```

Comment:
BTW, we really ought to return the actual enum back in REST and GRPC, it's useful, versus the boolean.

Reply:
Why? We do return in c#, but i'm guessing unity devs only care about if the findcloudlet was successful or not. Plus, returning FIND_NOTFOUND or FIND_UNKNOWN isn't very helpful in this case

Reply:
It's not a guarantee there will never have new states (right now, it is mostly just the 2 states, Unknown shouldn't happen outside a Unity AOT reflection error (link.xml, etc), for example), and a switch statement isn't much more difficult to handle.